src/ripple/rpc/handlers/AccountChannels.cpp

```diff
@@ -40,9 +40,12 @@ void addChannel (Json::Value& jsonLines, SLE const& line)
     jDst[jss::destination_account] = to_string (line[sfDestination]);
     jDst[jss::amount] = line[sfAmount].getText ();
     jDst[jss::balance] = line[sfBalance].getText ();
-    PublicKey const pk (line[sfPublicKey]);
-    jDst[jss::public_key] = toBase58 (TokenType::TOKEN_ACCOUNT_PUBLIC, pk);
-    jDst[jss::public_key_hex] = strHex (pk);
+    if (publicKeyType(line[sfPublicKey]))
+    {
+        PublicKey const pk (line[sfPublicKey]);
+        jDst[jss::public_key] = toBase58 (TokenType::TOKEN_ACCOUNT_PUBLIC, pk);
+        jDst[jss::public_key_hex] = strHex (pk);
+    }
```

Comment:
if this `publicKeyType` check fails, do we silently discard the data in `sfPublicKey`? Should we log a message or somehow notify the caller? I didn't review all the surrounding code, so this might already be dealt with in some other way.